src/video/uikit/SDL_uikitviewcontroller.m

```diff
@@ -528,28 +512,29 @@ - (void)keyboardWillShow:(NSNotification *)notification
     [self setKeyboardHeight:(int)kbrect.size.height];
 #endif
 
-    if (shouldStartTextInput) {
+    /* A keyboard hide transition has been interrupted with a show (keyboardWillHide has been called but keyboardDidHide didn't).
+     * since text input was stopped by the hide, we have to start it again. */
+    if (hidingKeyboard) {
         SDL_StartTextInput(window);
+        hidingKeyboard = NO;
     }
 }
 
-- (void)keyboardDidShow:(NSNotification *)notification
-{
-    showingKeyboard = NO;
-}
-
 - (void)keyboardWillHide:(NSNotification *)notification
 {
-    BOOL shouldStopTextInput = NO;
-
-    if (SDL_TextInputActive(window) && !showingKeyboard && !rotatingOrientation) {
-        shouldStopTextInput = YES;
-    }
-
     hidingKeyboard = YES;
     [self setKeyboardHeight:0];
 
-    if (shouldStopTextInput) {
+    /* When the user dismisses the software keyboard by the "hide" button in the bottom right corner,
+     * we want to reflect that on SDL_TextInputActive by calling SDL_StopTextInput...on certain conditions */
+    if (SDL_TextInputActive(window)
+        /* keyboardWillHide gets called when a hardware keyboard is attached,
+         * keep text input state active if hiding while there is a hardware keyboard.
+         * if the hardware keyboard gets detached, the software keyboard will appear anyway. */
+        && !SDL_HasKeyboard()
+        /* When the device changes orientation, a sequence of hide and show transitions are triggered.
+         * keep text input state active in this case. */
+        && !rotatingOrientation) {
         SDL_StopTextInput(window);
     }
```

Comment:
Upon further testing, having this also breaks password manager integrations:

https://github.com/user-attachments/assets/6a92fb21-f218-4762-b77a-aa4450ad450c

Notice how pressing the "Passwords" button in the keyboard hides the keyboard, which triggers the SDL application to stop text input. This is problematic because when the user selects their password from the keychain, the input is ignored since text input is not active.

I think it would be a better option to keep text input state completely manual rather than tied to the state of the keyboard, and do away with all these conditionals that are set in place here. @slouken thoughts? do other platforms (aka Android) stop text input when the user hides the keyboard?

Reply:
Yes, the expectation is that text input state is tied to the visibility of the on-screen keyboard, if there is no hardware keyboard attached.

Reply:
With the situation above, there is no possible way to enter the selected password without breaking that assumption, unless SDL can somehow bypass the "text input active" check when sending text events and do that on `textFieldTextDidChange` events, but I think that's too much of a hacky solution IMO.

Assuming that each application that starts text input has a path for stopping text input, it would be much easier to keep text input state manual and detached from the visibility of the on-screen keyboard, to allow integration with password managers etc.